C++/include/algorithm/dynamic_programming/README.md

```diff
@@ -44,3 +50,196 @@ Time               | Space
 _O(N<sup>3</sup>)_ | _O(N<sup>2</sup>)_
 
 where N is the number of matrices.
+
+## 2. Coin Change
+
+Given an unlimited amount of coins with different values, find the number of ways of making change for an given
+value using the coins.
+
+## Input
+- #### `coin`
+  Type: `vector<int> &`
```

Comment:
I'm not sure that it's necessary to include that coin is a reference (&). Perhaps better as just `vector<int>`.